### PR TITLE
add generic stream wrapper developed by icewind as 3rdparty dependency

### PR DESCRIPTION
**Pull request by schiesbn**
add generic stream wrapper developed by @icewind1991, it is already in use for the samba back-end and will be used for encryption as well.

### CODE REVIEW

**Comment by icewind1991** [comment]
I would also like to sometimes base the existing stream wrappers we have onto this

(this PR doesnt contain the code, only the composer.json)

File: composer.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
 		"symfony/routing": "~2.5",
 		"pimple/pimple": "~3.0",
 		"ircmaxell/password-compat": "1.0.*",
-		"nikic/php-parser": "~1.1"
+		"nikic/php-parser": "~1.1",
+		"icewind/Streams": "dev-master"
```
Comment (icewind1991) [inline composer.json]: please refer to a version instead (0.2.0)

**Comment by schiesbn** [comment]
@icewind1991 updated the version and added the code

File: icewind/streams/tests/Wrapper.php [hunk 2]
```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * Copyright (c) 2014 Robin Appelman <icewind@owncloud.com>
+ * This file is licensed under the Licensed under the MIT license:
+ * http://opensource.org/licenses/MIT
+ */
+
+namespace Icewind\Streams\Tests;
```
Comment (DeepDiver1975) [inline icewind/streams/tests/Wrapper.php]: tests have to be added to .gitignore - THX

Comment (schiesbn) [inline icewind/streams/tests/Wrapper.php]: done

**Comment by schiesbn** [comment]
did I still miss something? If I checkout this branch to my 3rdparty directory and try to load Icewind\Streams\wrapper i get a class not found.

**Comment by icewind1991** [comment]
you forgot to commit the changes to the composer autoload (and all the other composer files)

**Comment by schiesbn** [comment]
Everything should be in the PR now.

**Comment by DeepDiver1975** [comment]
... and rebase is required - this is a nightmare in here - let me know if you need support - I'm happy to help out

**Comment by schiesbn** [comment]
@DeepDiver1975 I removed the test files but I'm not sure how to resolve the merge conflicts without breaking something. Would be great if you could have a look at it. Thanks!

**Comment by DeepDiver1975** [comment]
taking over ....

**Comment by DeepDiver1975** [comment]
@schiesbn done

**Comment by schiesbn** [comment]
@DeepDiver1975 great, Thanks! So what's the next step? Can we merge it? Any reason not to do it?

**Comment by DeepDiver1975** [comment]
We should merge this branch as soon as the corresponding branch in core is ready to be merged.
So I guess this will take still some time.

We will have the rebasing fun quite a few times more until this happens.

**Pull request merged**